_posts/2019-12-17-zero-to-main-rust-1.md

```diff
@@ -0,0 +1,433 @@
+---
+title: "From zero to main(): Bare metal Rust"
+description: "Bringing an embedded Rust application from zero to main"
+author: james
+tags: [zero-to-main]
+---
+
+For the past thirty years or so, the choices of languages for embedded systems developers has been relatively slim. Languages like C++ and Ada have found a home in some niche areas, such as telecommunications and safety critical fields, while higher level languages like Lua, Python, and JavaScript have found a home for scripting and prototyping. But most developers working on bare metal systems have been using the same two languages as long as I can remember: Assembly and C.
+
+But not for no reason! Languages often make trade-offs to fit the needs of the developers working with them: an interpreter to allow for rapid iteration, a heap for ease of memory management, exceptions for simplifying control flow, etc. But these trade-offs come with a price: whether it is code size, RAM usage, low level control, power usage, latency, or determinism.
+
+<!-- excerpt start -->
+
+Since 2015, Rust has been redefining what it means to combine the best-in-class aspects of performance, correctness, and developer convenience into one language, without compromise.
```

Comment:
```suggestion
Since 2015, Rust has been redefining what it means to combine the best-in-class aspects of performance, correctness, and developer convenience into one language, without compromise. In this post, we bootstrap a rust environment on a cortex-M microcontroller and walk through some of the benefits of using rust on embedded.
```

Reply:
Suggestion: add a note about the goal of the post in the excerpt, as such ^